### PR TITLE
Fix empty basic auth

### LINKED ISSUE #1592: Traefik crash when failing to parse Basic Authentication information
Version: `Traefik version v1.3.0-rc1 built on 2017-05-05_02:03:01PM`

### Scenario
Starting traefik in docker and trying to route to another docker container using `labels`.
If incorrect (or in the test case empty) authentication information is used in the container I try to route to - traefik outputs:
`level=fatal msg="Error creating Auth: Error parsing Authenticator user: "`
and crashes.

### Expected outcome
If traefik fails to parse authentication information it should (probably) ignore the frontend mapping and not add the routing to the container.

### What did you see instead?
Traefik crashes causing all routing to stop working.

### Steps to reproduce?
`docker-compose up`

With the following compose file:
```yml
version: '3'

services:
  webgateway:
    image: traefik:1.3.0-rc1-alpine
    command:
      - --docker
      - --web
      - --logLevel=DEBUG
    ports:
      - "8080:8080"
      - "8000:80"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  error:
    image: nginx:alpine
    labels:
      - traefik.port=80
      - traefik.frontend.auth.basic=
```

### PR DESCRIPTION
**Pull request by emilevauge**
### Description

This PR fixes traefik crash when basic auth user is empty.

Fixes #1592

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -843,7 +844,7 @@ func (server *Server) buildDefaultHTTPRouter() *mux.Router {
 	return router
 }
 
-func parseHealthCheckOptions(lb healthcheck.LoadBalancer, backend string, hc *types.HealthCheck, hcConfig HealthCheckConfig) *healthcheck.Options {
+func parseHealthCheckOptions(lb healthcheck.LoadBalancer, backend string, hc *types.HealthCheck, hcConfig *HealthCheckConfig) *healthcheck.Options {
```
Comment (timoreimann) [inline server/server.go]: Do we guarantee on the call site that `hcConfig` is always != nil? Without that, we could run into a nil reference issue on line 852.

**Comment by timoreimann** [comment]
Hmm, how's the health check configuration related to basic auth?

I must be missing a piece of the puzzle... 🤔

File: server/server.go (hunk 1 above)
Comment (emilevauge) [inline server/server.go]: Oops, I was thinking the same and I just saw you too :) Fixed.

**Comment by emilevauge** [comment]
@timoreimann You are absolutely right. I just fixed this because it failed while I was writing the new tests and I thought it would be better to remove the pointer dereferencing calling `parseHealthCheckOptions`. WDYT?

**Pull request merged**